node/src/neighborhood/mod.rs

```diff
@@ -546,6 +547,21 @@ impl Neighborhood {
         }
     }
 
+    fn validate_database_min_hops_count(&self) {
```

Comment:
Hmmm. Is this really necessary? 

Think with me. How was the value stored in the actor's body set up? It happened in the actor system factory. Do you know what those actors use during their setup? The vast majority (if not all) of parameters that they need comes from the BootstrapperConfig whose existence is conditioned by this very moment. Ideally, the actors should just be fed by the relevant fields (some have their own whole configs there) in that aggregated config.

That means the Neighborhood when we're preparing it to start it up later should get the min_hops_count out of the BootstrapperConfig. (If it doesn't you need to go and change things so that it will. This pattern is pretty important.) That implies that this value here at the line 555 should be exactly the value that was finding itself in the BootstrapperConfig when it was finalized. 

Then ask yourself, what was that what assisted when the desired mins_hop_count was newly set to the database. Yes, again, the BootstrapperConfig on its own. Now you might see that both of those places that you test in the function here for equality got their values from exactly the same source. Then tell me why do you assert whether they are the same.

I don't see why 
a) it might have changed
b) why it is such a dangerous parameter that it stimulated you to write this code

I think we can stay calm even without this function so please delete it.

Reply:
I added this to make it safe for any changes that we may or may not do in the future.

If by any chance there are any side effects due to such changes, then the panic will remind us that we've broken the pattern somewhere.

I believe it makes our code more secure.

Reply:
Because we don't act like that with other parameters it still seems out of place here. I believe Dan and I wouldn't have done this or at least insisted to keep it.

I'd like you to discuss this with Dan because it is potentially needless code.

Reply:
Done 👍 

We're picking the value from db now.